components/char_collection/src/char_collection.rs

```diff
@@ -0,0 +1,828 @@
+// Copyright 2019 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+use std::{
+    boxed::Box,
+    char,
+    clone::Clone,
+    cmp::Ordering,
+    convert::Into,
+    error::Error,
+    hash::{Hash, Hasher},
+    iter::Iterator,
+    ops::Range,
+    vec::Vec,
+};
+
+#[derive(Copy, Clone, Debug, Eq)]
+pub struct CharRange {
+    low: char,
+    high: char,
+}
+
+impl CharRange {
+    // open_right
+    // would we want this to return a Option next time?
+    pub fn open_right(low: char, high: char) -> CharRange {
+        // nothing happens if this fails
+        let high: char = char::from_u32(high as u32 - 1).unwrap();
+        CharRange { low, high }
+    }
+    // closed
+    pub fn closed(low: char, high: char) -> CharRange {
+        // if low == '\u{0}' { // need way to handle this
+        //     // for now just leave alone
+        // }
+        CharRange { low, high }
+    }
+    // open
+    pub fn open(low: char, high: char) -> CharRange {
+        // this is repeated here
+        let low: char = char::from_u32(low as u32 + 1).unwrap();
+        let high: char = char::from_u32(high as u32 - 1).unwrap();
+        CharRange { low, high }
+    }
+    // open_left
+    pub fn open_left(low: char, high: char) -> CharRange {
+        // this is repeated here
+        let low: char = char::from_u32(low as u32 + 1).unwrap();
+        CharRange { low, high }
+    }
```

Comment:
This seems like a scary amount of `unwrap` rather than making them falliable via `Result`.